pkg/api/v1beta1/types.go

```diff
@@ -1561,3 +1564,100 @@ type GlusterfsVolumeSource struct {
 	// the Glusterfs volume to be mounted with read-only permissions
 	ReadOnly bool `json:"readOnly,omitempty" description:"Glusterfs volume to be mounted with read-only permissions"`
 }
+
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
```

Comment:
@bgrant0607 
suggests:
 - don't inline
 - Have both the spec.privileged and spec.securityContext.privileged; and likewise for capabilities
  - post-defaulting
 - In validation, reject conflicting values, otherwise default one from the other.
 - see validation of ResourceList for an example.

This shows how we intend to do it post v1, but supports older config files.